tasks/framework-tools/src/frameworkSyncToProject.js

```diff
@@ -1,6 +1,102 @@
-// TODO:
+#!/usr/bin/env node
+/* eslint-env node, es6 */
 
-// Add the deps
-// Run install
-// Copy/ build the files when they change
-// Remove them when user cancels
+/**
+ * - add the deps and install
+ * - build and copy over files
+ * - remove them when user cancels
+ */
+
+const chokidar = require('chokidar')
+const execa = require('execa')
+
+// add the deps and install
+
+const addFrameworkDepsToProject = () => {
+  try {
+    execa.sync('yarn project:deps', {
+      shell: true,
+      stdio: 'inherit',
+      env: {
+        RWJS_CWD: process.env.RWJS_CWD,
+      },
+    })
+  } catch (e) {
+    console.error(
+      'Error: Could not add Redwood Framework dependencies to project'
+    )
+    console.error(e)
+    process.exit(1)
+  }
+}
+
+const runYarnInstall = () => {
+  try {
+    execa.sync('yarn install', {
+      cwd: process.env.RWJS_CWD,
+      shell: true,
+      stdio: 'inherit',
+    })
+  } catch (e) {
+    console.error('Error: Could not run `yarn install`')
+    console.error(e)
+    process.exit(1)
+  }
+}
+
+// build and copy over files
+
+const buildRedwoodFramework = () => {
+  try {
+    execa.sync('yarn build:clean && yarn build:js', {
+      shell: true,
+      stdio: 'inherit',
+    })
+  } catch (e) {
+    console.error('Error: Could not build Redwood Framework')
+    console.error(e)
+    process.exit(1)
+  }
+}
+
+const copyFrameworkPackages = () => {
+  try {
+    execa.sync('yarn project:copy', {
+      shell: true,
+      stdio: 'inherit',
+      env: {
+        RWJS_CWD: process.env.RWJS_CWD,
+      },
+    })
+  } catch (e) {
+    console.error('Error: We could not copy Redwood Framework packages')
+    console.error(e)
+    process.exit(1)
+  }
+}
+
+//
+
+try {
+  addFrameworkDepsToProject()
+  runYarnInstall()
+  buildRedwoodFramework()
+  copyFrameworkPackages()
+} catch (e) {
+  console.log()
+}
+
+// TODO we might be able to get close to pattern matching by doing...
+//
+// chokidar
+//   .watch(frameworkFiles, options)
+//   .on('change', (path) => {
+//     switch (true) {
+//       case depsChanged(path):
+//         ...
+//       case frameworkChanged(path):
+//         ...
+//       default:
+//         ...
+//     }
+//   })
```

Comment:
Take a look at https://github.com/redwoodjs/redwood/blob/41aa291e520139141336e418dab999ac55a3d456/packages/api-server/src/watch.ts

It's a fairly clean implementation of chokidar.

I think what we should do is:
1. watch the `packages/**` folders, whilst ignoring things like test files
2. during the `on('ready')` event, run the initial copy the deps (and keep the packages object in memory), run yarn install, and copy of the package files.

3a). Copying the files: then if a change is detected in a particular package, run `yarn build` in that packages folder, and copy over the files of that package. (`packagesFileList` allows you to specify an array of packages that it should operate against.)

3b). Copying the deps: If the change is in a `package.json` file determine if the contributor has installed a new package by comparing the deps (that we keep in memory from step 2), and a newly generated list of complete deps., warn them that they need to run yarn install.

Reply:
Thanks! I'll reference that for sure; and starting to draw the logic here

![image](https://user-images.githubusercontent.com/32992335/124793414-f6a43600-df1b-11eb-8070-fa1c07b9e6fd.png)